### ISSUE TITLE
DLL load failed

### ISSUE THREAD

**Issue by ThomasTommy** [CoolProp/CoolProp#1156]
hi,I'm a student interested in CoolProp.I have installed it successful by run into _pip install coolprop_
But,it can't work well.
Error: from .CoolProp import AbstractState
ImportError: DLL load failed: can't find specified module
I just konw these informations.
Thank you!!!

**Comment by ibell** [comment]
What version of python do you have?  What do you get if you do something like this at the console:

```
C:\Users\Belli>python
Python 2.7.11 |Continuum Analytics, Inc.| (default, Feb 16 2016, 09:58:36) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
Anaconda is brought to you by Continuum Analytics.
Please check out: http://continuum.io/thanks and https://anaconda.org
```

It should "just work".  I've not heard of this issue before.

**Comment by ThomasTommy** [comment]
Python3.5.1
Windows 8

**Comment by ibell** [comment]
I think it must be a windows 8 issue.

But to be sure, do a few pip uninstall Coolprop until you can't uninstall
anymore.  And then reinstall.

**Comment by ThomasTommy** [comment]
Ok，thank you!!!

**Comment by ibell** [comment]
Did that work?

**Comment by ibell** [comment]
Bump.  Did that work?

**Comment by ashshrma** [comment]
Hello, I'm facing the same issue. 
I have tried uninstalling and re-installing (through pip and the wheel from pypi). Nothing seems to work. 
I am running Python 3.7.4 - 32 bit on Windows 7.
Any ideas about resolution.?

**Comment by ibell** [comment]
Can you use 64-bit Python?  I have no problems with 64-bit python 3.7 with
conda and coolprop.  Are you using conda? Strongly recommended.

**Comment by ashshrma** [comment]
Thank you for the response. 
I am using virtualenv, tried installing on the native python installation as well. Fails with the same error. 
Also, to bring to your notice, this seems to be a machine specific error. The exact same set-up in on another machine works without a problem.

**Comment by ibell** [comment]
You might also want to try to uninstall CoolProp

Also, you might want to check your PATH, maybe something in there is
messing things up.

**Comment by ashshrma** [comment]
Uninstalling works without a problem. I have tried removing the path entries, retained only the basic system ones, does not make any difference.

**Comment by ibell** [comment]
I'll bet you have a funny standard library in your path.  And it finds that
and then it can't load.

I think if you were to use an anaconda environment it would work.  Maybe...

**Comment by ashshrma** [comment]
I'm guessing so too..
I am going to live without CoolProp in this system and use the other one where it is working fine. 
I will post back (if and when) I try the anaconda env and it works.

Thank you for your time and feedback!

**Comment by koreantiger** [comment]
Hi there, 
I am getting similar error and i am using Anaconda3.6 windows 10. During the installation also i got this warning "distributed 1.21.8 requires msgpack, which is not installed". I dont know whether it is the reason that i get the similar error (DLL load failed).

**Comment by ibell** [comment]
Can you please show what you tried to do that generated this error?

**Comment by Arkamus** [comment]
Hi,

I also had problem with:

Error: from .CoolProp import AbstractState
ImportError: DLL load failed: can't find specified module

In my case, this problem occured for the first time after installing CoolProp 6.4 on my new laptop. I uninstalled it,  installed CoolProp 6.3 and everything seems to work fine again.

**Comment by ibell** [comment]
There seems to be some issues with anaconda and the 6.4.0 release having to
do with standard libraries.  Please try to install the c++ redistributable
from Microsoft.

**Comment by wdeason07** [comment]
Having this issue too - which c++ redistributable do I need to reinstall?

**Comment by ibell** [comment]
See also #1940

**Comment by ibell** [comment]
Thanks @wdeason07 !

**Comment by HermanTheGerman69** [comment]
Hey guys,

I have the exact same problem.
I am studying at uni and we have certain programs that we use.
One of them is Spyder. Now we have to use CoolProp on Spyder, but I keep getting a "DLL load failed" error. I tried uninstalling CoolProp and installing it again but that didn't work. I also checked the Path but that wasn't the problem either.
(Btw I installed CoolProp with an Anaconda Prompt)
I am not very skilled with computers, so does anyone have an idea what I could do?

**Comment by HermanTheGerman69** [comment]
Thanks to the comment of @wdeason07 I got it to work! Installing the Microsoft C++ 2019 Redistributable made it work ...   Microsoft killed my whole weekend (again... :/ )

**Comment by ibell** [comment]
Yup this is one where ask the question early, get the answer early
hopefully.  Haven't given up on fixing this issue yet.  Proving to be
tricky.

**Comment by ibell** [comment]
Please also see a preferred solution of installing the python package ``msvc_runtime``: #1940 #1942

**Comment by ibell** [comment]
Closing as part of an issue-tracker cleanup — this issue has had no activity in over two years.

*To be transparent: this comment and the close action are being performed by Claude (Anthropic's AI coding assistant), running on behalf of @ibell. Ian reviewed and approved the bulk-close criteria; I am the one executing the action.*

If you would like this issue reconsidered, please feel free to **re-open it** — a fresh comment with current behaviour against the latest CoolProp release is the most useful way to revive the discussion.

— Claude (acting for @ibell)
